src/lib.rs

```diff
@@ -1,36 +1,74 @@
-use std::ptr::null_mut;
-use std::os::raw::c_int;
+mod libxenevtchn;
+extern crate xenevtchn_sys;
+use libxenevtchn::LibXenEvtchn;
+use log::debug;
+use std::convert::TryInto;
 use std::io::Error;
+use std::os::raw::c_int;
+use std::ptr::null_mut;
+use xenevtchn_sys::{evtchn_port_t, xenevtchn_handle, xenevtchn_port_or_error_t};
 
 #[derive(Debug)]
 pub struct XenEventChannel {
-    handle: *mut xenevtchn_sys::xenevtchn_handle,
-    fd: c_int,
+    handle: *mut xenevtchn_handle,
     bind_port: c_int,
+    libxenevtchn: LibXenEvtchn,
 }
 
 impl XenEventChannel {
+    pub fn new(domid: u32, evtchn_port: u32) -> Result<Self, Error> {
+        let handle = unsafe { xenevtchn_sys::xenevtchn_open(null_mut(), 0) };
```

Comment:
You should use `self.libxenevtchn` in all calls that originally used `xenevtchn_sys`. If we don't replace all of those calls, the bindings will still be linked with libxenevtchn which we would like to avoid.